### PR TITLE
[feature](array function) add support for array_enumerate_uniq

### PR DESCRIPTION
**Pull request by bobhan1**
# Proposed changes

Issue Number: #17448 

## Problem summary

add support for `array_enumerate_uniq()`

## Checklist(Required)

* [x] Does it affect the original behavior
* [x] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
@eldenmoon

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run p0

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.95 seconds
 stream load tsv:          460 seconds loaded 74807831229 Bytes, about 155 MB/s
 stream load json:         39 seconds loaded 2358488459 Bytes, about 57 MB/s
 stream load orc:          74 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230308073754_clickbench_pr_110629.html

**Comment by bobhan1** [comment]
I run the regression test locally and it failed at an irrelevant test suit:
```
2023-03-08 16:04:13.763 INFO [main] (RegressionTest.groovy:277) - Failure suites:
/home/baohan/projects/doris/regression-test/suites/query_p0/sql_functions/array_functions/sql/q03.sql: group=default,p0, name=q03
```
`/doris/regression-test/suites/query_p0/sql_functions/array_functions/sql/q03.sql`:
```
DROP TABLE IF EXISTS array_insert_select_test;
CREATE TABLE IF NOT EXISTS array_insert_select_test (id int, c_array array<int(11)>) ENGINE = Olap DUPLICATE KEY(id) DISTRIBUTED BY HASH(id) BUCKETS 1 PROPERTIES('replication_num' = '1');
insert into array_insert_select_test select k1, collect_list(k3) from test_query_db.test group by k1;
select c_array from array_insert_select_test;
```
Should I do the rebase?

**Comment by eldenmoon** [comment]
@bobhan1 you could ignore `array_functions/sql/q03.sql` while run local, since the query rely on a table from load related regression test case

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
Does the number of checks depend on the cheklist above?

**Comment by bobhan1** [comment]
run p1

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run p0

File: be/src/vec/functions/array/function_array_enumerate_uniq.cpp [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                _execute_number<ColumnFloat64>(data_columns, *offsets, null_map, dst_values);
+            } else if (which.is_date()) {
+                _execute_number<ColumnDate>(data_columns, *offsets, null_map, dst_values);
+            } else if (which.is_date_time()) {
+                _execute_number<ColumnDateTime>(data_columns, *offsets, null_map, dst_values);
+            } else if (which.is_date_v2()) {
+                _execute_number<ColumnDateV2>(data_columns, *offsets, null_map, dst_values);
+            } else if (which.is_date_time_v2()) {
+                _execute_number<ColumnDateTimeV2>(data_columns, *offsets, null_map, dst_values);
+            } else if (which.is_decimal128()) {
```
[the remark this answers is not in the document]
Comment (Yukang-Lian) [reply be/src/vec/functions/array/function_array_enumerate_uniq.cpp:150]: all right.

**Review by Yukang-Lian — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Comment by bobhan1** [comment]
I wonder why the following checks are not triggered.

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**